conans/client/tools/cmake_export.py

```diff
@@ -0,0 +1,60 @@
+import os
+from itertools import chain
+
+def _patch_file(cmake,conanfile,fname):
+    with open(fname,"r") as f:
+        filestr = f.read()
+        
+    pf = cmake.definitions.get("CMAKE_INSTALL_PREFIX")
+    replstr = "${CONAN_%s_ROOT}" % conanfile.name.upper()
+    
+    newfilestr = filestr.replace(pf,replstr)
+    with open(fname,"w") as f:
+        f.write(newfilestr)
+        
+def patch_cmake_config_abs_paths(cmake):
+    """
+    changes references to the absolute path of the installed package in 
+    exported cmake config files to the appropriate conan variable. This makes 
+    most (sensible) cmake config files portable.
+    
+    For example, if a package foo installs a file called "fooConfig.cmake" to 
+    be used by cmake's find_package method, normally this file will contain 
+    absolute paths to the installed package folder, for example it will contain
+    a line such as:
+        
+        SET(Foo_INSTALL_DIR /home/developer/.conan/data/Foo/1.0.0/...)
+        
+    This will cause cmake find_package() method to fail when someone else 
+    installs the package via conan. 
+    
+    This function will replace such mentions to
+        
+        SET(Foo_INSTALL_DIR ${CONAN_FOO_ROOT})
+        
+    which is a variable that is set by conanbuildinfo.cmake, so that find_package()
+    now correctly works on this conan package. 
+    
+    If the install() method of the CMake object in the conan file is used, this
+    function should be called _after_ that invocation. For example:
+        
+        def build(self):
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
+            cmake.install()
+            
+            patch_cmake_config_abs_paths(cmake)
```

Comment:
Just wondering, and I also think that making it explicit, like this, might be the best.
But, what if it is done automatically? Because if I understood correctly, it will always be a problem for all generated config files?

Reply:
yes, it has to be done for all config files, but from what I've seen of conan, it is usually the case that steps like this are performed manually. Another possibility is to make it an argument of the cmake.install() method? e.g. cmake.install(patch_config_files=True) or something.